check_api/src/main/java/com/google/errorprone/FileReporter.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2018 The Error Prone Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.errorprone;
+
+import com.sun.source.util.TaskEvent;
+import com.sun.source.util.TaskListener;
+import com.sun.tools.javac.util.DiagnosticSource;
+import com.sun.tools.javac.util.JCDiagnostic;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.time.ZonedDateTime;
+import org.apache.commons.text.StringEscapeUtils;
+
+/**
+ * Generates a parseable report file.
+ */
+public class FileReporter implements TaskListener {
+
+  PrintWriter pw;
+
+  public FileReporter(ErrorProneOptions options) {
+    try {
+      final String reportFile = options.getReportFile();
+      if (reportFile != null && !reportFile.isEmpty()) {
+        pw = new PrintWriter(new FileWriter(options.getReportFile()));
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void started(final TaskEvent e) {
+    if (e.getKind() == TaskEvent.Kind.COMPILATION) {
+      if (pw != null) {
+        pw.println("<?xml version=\"1.0\" encoding=\"utf-8\"?>");
+        pw.println("<errorprone timestamp=\"" + ZonedDateTime.now() + "\">");
```

Comment:
This should be formatted, ideally with an ISO standard.

Reply:
It is already: `ZonedDateTime#toString()` produces an ISO 8601 timestamp.

Reply:
Relying on a `toString` implementation feels a bit fragile?  I might have used a `DateFormat` with the format specifier `"yyyy-MM-dd'T'HH:mm:ss.SSS"` just to make it more obvious?